watertap/unit_models/tests/test_cstr.py

```diff
@@ -0,0 +1,446 @@
+        assert value(
+            model.fs.unit.outlet.conc_mol_comp[0, "SodiumAcetate"]
+        ) == pytest.approx(79.683910, rel=1e-5)
+        assert value(model.fs.unit.outlet.conc_mol_comp[0, "Ethanol"]) == pytest.approx(
+            79.683910, rel=1e-5
+        )
+        assert value(model.fs.unit.outlet.temperature[0]) == pytest.approx(
+            304.0856, rel=1e-5
+        )
+        assert value(model.fs.unit.outlet.pressure[0]) == pytest.approx(
+            101325, rel=1e-5
+        )
+
+        assert not model.fs.unit.inlet.flow_vol[0].fixed
+        assert not model.fs.unit.inlet.conc_mol_comp[0, "H2O"].fixed
+        assert not model.fs.unit.inlet.conc_mol_comp[0, "NaOH"].fixed
+        assert not model.fs.unit.inlet.conc_mol_comp[0, "EthylAcetate"].fixed
+        assert not model.fs.unit.inlet.conc_mol_comp[0, "SodiumAcetate"].fixed
+        assert not model.fs.unit.inlet.conc_mol_comp[0, "Ethanol"].fixed
+
+        assert not model.fs.unit.inlet.temperature[0].fixed
+        assert not model.fs.unit.inlet.pressure[0].fixed
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_costing(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(dynamic=False)
+
+        m.fs.props_ASM1 = ASM1ParameterBlock()
+        m.fs.ASM1_rxn_props = ASM1ReactionParameterBlock(
+            property_package=m.fs.props_ASM1
+        )
+
+        m.fs.unit = CSTR(
+            property_package=m.fs.props_ASM1, reaction_package=m.fs.ASM1_rxn_props
+        )
+
+        m.fs.unit.inlet.flow_vol[0].set_value(1.2199)
+        m.fs.unit.inlet.alkalinity[0].set_value(4.5102)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_I"].set_value(0.061909)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_S"].set_value(0.012366)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_I"].set_value(1.4258)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_S"].set_value(0.090508)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_BH"].set_value(2.8404)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_BA"].set_value(0.20512)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_P"].set_value(0.58681)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_O"].set_value(0.00036092)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_NO"].set_value(0.012424)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_NH"].set_value(0.0076936)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_ND"].set_value(0.0019068)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_ND"].set_value(0.0053166)
+
+        m.fs.unit.inlet.temperature[0].set_value(308.15)
+        m.fs.unit.inlet.pressure[0].set_value(84790.0)
+
+        m.fs.unit.volume[0].fix(1000)
+
+        m.fs.costing = WaterTAPCosting()
+
+        m.fs.unit.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
+        m.fs.costing.cost_process()
+        m.fs.costing.add_LCOW(m.fs.unit.control_volume.properties_out[0].flow_vol)
```

Comment:
Can you also `assert_units_consistent`? I am skeptical about the units in the cost equation.